hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -374,21 +374,23 @@ private void processQueuedBlocksForInstant(Deque<HoodieLogBlock> logBlocks, int
       LOG.info("Number of remaining logblocks to merge " + logBlocks.size());
       // poll the element at the bottom of the stack since that's the order it was inserted
       HoodieLogBlock lastBlock = logBlocks.pollLast();
-      switch (lastBlock.getBlockType()) {
-        case AVRO_DATA_BLOCK:
-          processDataBlock((HoodieAvroDataBlock) lastBlock, keys);
-          break;
-        case HFILE_DATA_BLOCK:
-          processDataBlock((HoodieHFileDataBlock) lastBlock, keys);
-          break;
-        case DELETE_BLOCK:
-          Arrays.stream(((HoodieDeleteBlock) lastBlock).getKeysToDelete()).forEach(this::processNextDeletedKey);
-          break;
-        case CORRUPT_BLOCK:
-          LOG.warn("Found a corrupt block which was not rolled back");
-          break;
-        default:
-          break;
+      if (lastBlock != null) {
```

Comment:
Can you help me understand why we need this change?

Reply:
During EMR/S3 testing, I encountered NPE in this block. Looking more into it, ArrayDeque is not thread safe (though  my test wasn't doing any multi writer tests) and `isEmpty()` check is not fool proof. That is, it can say it is not empty and still have poll() returning null. Haven't got a chance to dig how this case is getting exposed in my test case, but for now closing the gap and avoiding the crash by doing an additional check.

Reply:
hmmm. Can you update the description of the patch then, wrt this fix. 
Also, can we add in a test for this fix. May be it could affect non-inline path as well.

Reply:
Spent some more time on this, but couldn't recreate the problem. The one difference now is, the ExternalSpillableMap throwing ArithmeticException is fixed now which wasn't the case when I tested on EMR. So, revering this lastBlock null check for now, until i am able to recreate the problem and root cause it.